DigitalLearningSolutions.Web/ViewModels/Register/RegisterDelegateByCentre/WelcomeEmailViewModel.cs

```diff
@@ -22,7 +24,17 @@ public WelcomeEmailViewModel(DelegateRegistrationByCentreData data)
         public int? Month { get; set; }
         public int? Year { get; set; }
         public bool ShouldSendEmail { get; set; }
-        public DateValidator.ValidationResult? DateValidationResult { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            if (!ShouldSendEmail)
+            {
+                return new List<ValidationResult>();
+            }
+
+            var dateValidationResult = DateValidator.ValidateDate(Day, Month, Year, "Email delivery date", true);
+            return DateValidator.ToValidationResultList(dateValidationResult, "Day", "Month", "Year");

```

Comment:
If these always go hand-in-hand would it make sense to always do these together in DateValidator?

Reply:
I thought since it is a validator, ValidateDate should simply validate a date and return an object which could easily tell you the validity and error. Forcing it to always be a list of ValidationResult seems unexpected. This way its more flexible? Not sure if one way is better than the other?

Reply:
ToValidationResultList moved into DateValidationResult as discussed